gpMgmt/bin/pythonSrc/PyGreSQL-4.0/pgmodule.c

```diff
@@ -42,6 +42,71 @@
 
 #include <Python.h>
 
+#if PY_MAJOR_VERSION >= 3
+
+/***** Python 3 *****/
+
+/* Strings */
+
+#define PyString_Check PyUnicode_Check
+#define PyString_FromString PyUnicode_FromString
+#define PyString_FromStringAndSize PyUnicode_FromStringAndSize
+#define PyString_AsString PyUnicode_AsUTF8
+
+#define PyStr_AsUTF8String PyUnicode_AsUTF8String // returns PyBytes
+#define PyStr_AsUTF8AndSize PyUnicode_AsUTF8AndSize
+
+/* Ints */
+
+#define PyInt_Check PyLong_Check
+#define PyInt_FromString PyLong_FromString
+#define PyInt_FromLong PyLong_FromLong
+#define PyInt_AsLong PyLong_AsLong
+
+/* Module init */
+
+#define MODULE_INIT_FUNC(name) \
+	PyMODINIT_FUNC PyInit_ ## name(void); \
+	PyMODINIT_FUNC PyInit_ ## name(void)
+
+#else /* if PY_MAJOR_VERSION >= 3 */
+
+/***** Python 2 *****/
+
+/* Strings */
+
+#define PyStr_AsUTF8String(str) (Py_INCREF(str), (str))
+#define PyStr_AsUTF8AndSize(pystr, sizeptr) \
+	((*sizeptr=PyString_Size(pystr)), PyString_AsString(pystr))
+
+/* Floats */
+
+#define PyFloat_FromString(str) PyFloat_FromString(str, NULL)
```

Comment:
How does this work if `PyFloat_FromString` is already defined in `Python.h`? Should we do the opposite to reduce the diff?
```c
#if PY_MAJOR_VERSION >= 3
#define PyFloat_FromString(str, pend) PyFloat_FromString(str)
#endif
```